sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -157,15 +145,17 @@ export class ServiceBusClient {
     );
 
     if (receiveMode === "peekLock") {
-      return new ReceiverImpl<ReceivedMessageWithLock>(clientEntityContext, receiveMode, {
-        ...options,
-        retryOptions: options?.retryOptions ?? this._clientOptions.retryOptions
-      });
+      return new ReceiverImpl<ReceivedMessageWithLock>(
+        clientEntityContext,
+        receiveMode,
+        this._clientOptions.retryOptions!
```

Comment:
Do we **have** to rely on using ! here?

I understand that in the client constructor we ensure that `retryOptions` is not null/undefined
But in `receiver.ts`, when you use the retryOptions, you do have a guard for options being null/undefined `this._retryOptions?.timeoutInMs`

So, what do we gain by adding `!` here?

I suggest that we update the constructor of `ReceiverImpl` to take the retry options as an optional parameter. You can default it to `{}` in the constructor and use `this._retryOptions.timeoutInMs` instead of `this._retryOptions?.timeoutInMs`

Reply:
Updated as per your suggestions.